test/dict.cpp

```diff
@@ -21,11 +21,13 @@ object new_dict()
 object data_dict()
 {
     dict tmp1;
-    tmp1["key1"] = "value1";
 
     dict tmp2;
     tmp2["key2"] = "value2";
     tmp1[1] = tmp2;
+
+    tmp1["key1"] = "value1";
+
```

Comment:
Just for avoidance of doubt: this reflects the fact that in current versions of Python the order reported (notably when printing a dict) corresponds to the insertion order, yes ?

Reply:
Yes! If we want to print out the full dictionary, we have to make sure the output will be the same across Python versions. I'm not sure what ordering Python 2 is using (by hash, perhaps), but Python 3 is using insertion order.

Reply:
Right. So if ever we find the order to vary (for older Python versions, where the order isn't mandated yet), we need to rethink how to print the dict...

Reply:
I think I tested each case on my local copy of Python2, and the order seemed fairly stable. Hopefully the same shows up in the CI.